src/Utilities/PrettyType.hpp

```diff
@@ -705,4 +706,31 @@ template <typename T>
 std::string short_name() {
   return detail::extract_short_name(typeid(T).name());
 }
+
+namespace detail {
+template <typename T, typename = std::void_t<>>
+struct name_helper {
+  static std::string name() { return pretty_type::short_name<T>(); }
+};
+
+template <typename T>
+struct name_helper<T, std::void_t<decltype(T::name())>> {
+  static std::string name() { return T::name(); }
+};
+}  // namespace detail
+
+/*!
+ * \ingroup PrettyTypeGroup
+ * \brief Return the result of the `name()` member of a class. If a class
+ * doesn't have a `name()` member, call `pretty_type::short_name<T>()` instead.
+ *
+ * \warning Do not use this inside the `name()` member of a DataBox tag. This
+ * can lead to recursion as `pretty_type::name<Tag>()` will call the `name()`
+ * member of the tag. A DataBox tag should only specify a `name()` member if the
+ * name is different than `pretty_type::short_name<Tag>()`
```

Comment:
I think remove the mention of tags. The recursion issue is independent of DataBox things. Just warn of the possible recursion when using this inside a name function.

Reply:
Done! :)